services/drupal/web/modules/custom/epa_media/epa_media.install

```diff
@@ -5,3 +5,131 @@
 function epa_media_update_9000() {
   \Drupal::database()->query("UPDATE {file_managed} SET filemime = 'application/json' WHERE filename LIKE '%.json' AND filemime = 'application/octet-stream';");
 }
+
+/**
+ * Move all json and xml "Document" media entities to be "Other" media bundle
+ */
+function epa_media_update_9001(&$sandbox) {
+  $database = \Drupal::database();
+
+  if (!isset($sandbox['total'])) {
+    // Find all "Document" media entities that have a .json or .xml file.
+    $media_ids = $database
+      ->query("SELECT entity_id
+        FROM {media__field_media_file}
+        WHERE field_media_file_target_id IN (
+	          SELECT fid
+	          FROM file_managed
+	          WHERE filename like '%.json' OR filename like '%.xml'
+        )
+        AND
+        bundle = 'document'")
+      ->fetchCol('entity_id');
+
+    $sandbox['total'] = count($media_ids);
+    $sandbox['current'] = 0;
+  }
+
+  // Do not continue if no media items are found.
+  if (empty($sandbox['total'])) {
+    $sandbox['#finished'] = 1;
+    return t('No Media entities to be processed.');
+  }
+
+  // Arbitrary limit to process 50 at a time.
+  $ids = array_slice($media_ids, $sandbox['current'], 50);
+
+  foreach ($ids as $id) {
+    // Update base tables with those media_ids
+    $database
+      ->query("UPDATE {media} AS m
+        INNER JOIN {media_field_data} AS mfd ON m.mid = mfd.mid
+        INNER JOIN {media_revision} AS mr ON m.mid = mr.mid
+        INNER JOIN {media_field_revision} AS mfr ON m.mid = mfr.mid
+        SET
+	          m.bundle = 'other',
+	          mfd.bundle = 'other',
+	          mr.bundle = 'other',
+	          mfr.bundle = 'other'
+        WHERE m.mid = :id", [':id' => $id]);
+
```

Comment:
This could be re-written to be something like this in order to do this update as a single query (not tested, so please test):
```
 $database
       ->query("UPDATE {media} AS m
         INNER JOIN {media_field_data} AS mfd ON m.mid = mfd.mid
         INNER JOIN {media_revision} AS mr ON m.mid = mr.mid
         INNER JOIN {media_field_revision} AS mfr ON m.mid = mfr.mid
         SET
 	          m.bundle = 'other',
 	          mfd.bundle = 'other',
 	          mr.bundle = 'other',
 	          mfr.bundle = 'other'
         WHERE m.mid IN (
             SELECT entity_id
             FROM {media__field_media_file} AS fmf 
               LEFT JOIN {file_managed} AS fm ON fm.fid = fmf.field_media_file_target_id 
             WHERE filename like '%.json' OR filename like '%.xml'
               AND bundle = 'document')");    
```

You should be able to follow a similar pattern for the rest of the queries, too.

Reply:
I've updated the queries to now use INs and passing 50 IDs at a time vs. individual queries. I'll test more today once I get the database working locally to see if there's any issue running all the IDs through it at once.